parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -1743,8 +1743,12 @@ public OffsetIndex readOffsetIndex(ColumnChunkMetaData column) throws IOExceptio
 
   @Override
   public void close() throws IOException {
+    close(true);
+  }
+
+  public void close(boolean closeFileInputStream) throws IOException {
```

Comment:
Is it better to explicitly add a `detachFileInputStream` method to set `f` to null so we don't need to hack the `close` here?

cc @gszadovszky

Reply:
thanks for your suggestion, but are you comfortable removing the `final` modifier for `f`?

```
protected final SeekableInputStream f;
```

Reply:
+1

Reply:
updated

Reply:
Thanks for the update! It would be good to reflect the change in the PR title.

Reply:
PR title is updated too

Reply:
Merged. Thanks @pan3793 and @gszadovszky!